src/java.base/share/classes/java/util/TimeZone.java

```diff
@@ -293,7 +293,11 @@ public String getID()
     /**
      * Sets the time zone ID. This does not change any other data in
      * the time zone object.
+     * @implSpec The default implementation throws a
+     * {@code NullPointerException} if {@code ID} is {@code null}
      * @param ID the new time zone ID.
+     * @throws NullPointerException This method may throw a
+     * {@code NullPointerException} if {@code ID} is {@code null}
```

Comment:
Is this needed given it is covered in the implSpec as it seems redundant

Reply:
In the CSR, Joe brought up the point that since _Time Zone_ is a non final class, and these instance methods can be overridden, there exists the possibility of an external subclass that overrides the method and does **not** throw an NPE. That is why in the @throws: section, the wording is weakened from _will_ throw an NPE to _may._

The **@ImplSpec** is then added on to state that Time Zone's default implementation of the particular method _does_ indeed throw an NPE.

Reply:
OK, fair enough, let's get the CSR moving forward to make sure the proposed wording addresses Joe's concern.

Reply:
Agreed, was planning to finalize the CSR before the end of the day, and will adjust this accordingly if Joe has concerns